File: Week_01/G20200343040015/merge-sorted-array.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
    if(nums1.length>m) nums1.splice(m);
    if(nums1.length>n) nums2.splice(n);
    Array.prototype.push.apply(nums1,nums2);
    nums1 = nums1.sort((a,b) =>a-b);
}

File: Week_01/G20200343040105/MergeSortedArray.cpp
//88. 合并两个有序数组

//从m+n开始从后往前归并排序

class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {

        int p1 = m - 1;
        int p2 = n - 1;
        int p = m + n - 1;
        while(p1 >= 0 && p2 >= 0)
        {
            nums1[p--] = (nums1[p1] > nums2[p2]) ? nums1[p1--] : nums2[p2--];
        }
        while(p2 >= 0)
        {
            nums1[p--] = nums2[p2--];
        }
        
    }
};
//时间复杂度 O(n) 空间复杂度 O(1)